master/internal/configpolicy/utils.go

```diff
@@ -81,11 +80,19 @@ func ValidateExperimentConfig(
 		return err
 	}
 
+	// Warn the user when fields specified in workspace config policies overlap with global config
+	// policies (since these fields will be overridden by the respective fields in the global
+	// policies).
+	var globalConstraints *model.Constraints
 	if globalConfigPolicies != nil {
-		checkAgainstGlobalConfig[model.Constraints](globalConfigPolicies.Constraints, cp.Constraints, "invalid constraints")
-		checkAgainstGlobalConfig[expconf.ExperimentConfig](
-			globalConfigPolicies.InvariantConfig, cp.InvariantConfig, InvalidExperimentConfigPolicyErr,
-		)
```

Comment:
Why are these function calls removed?

Reply:
Since we now need the unmarshaling logic at the beginning of `checkAgainstGlobalConfig` in `ValidateExperimentConfig` and `ValidateNTSCConfig`, I moved the unmarshaling logic one step higher and removed the function `checkAgainstGlobalConfig` since it called `configPolicyConflict` (renamed `warnConfigPolicyOverlap`) immediately after unmarshaling the global config policies and then returned. Great callout though! as I realized that I passed the wrong param into `warnConfigPolicyOverlap` (a string instead of an the expconf.ExperimentConfig). Since the function has a config-policy specific name, I added some type checks to the input parameters to make sure that we are passing in either invariant configs or constraints!